server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -466,32 +472,111 @@ public synchronized void verifyIndexMetadata(IndexMetadata metadata, IndexMetada
         }
     }
 
+
+    private CompletableFuture<IndexShard> createShard(ClusterState originalState,
+                                                      IndexService indexService,
+                                                      ShardRouting shardRouting,
+                                                      Consumer<ShardId> globalCheckpointSyncer,
+                                                      RetentionLeaseSyncer retentionLeaseSyncer,
+                                                      @Nullable Iterator<TimeValue> backoff) {
+        try {
+            IndexShard indexShard = indexService.createShard(shardRouting, globalCheckpointSyncer, retentionLeaseSyncer);
+            return CompletableFuture.completedFuture(indexShard);
+        } catch (ShardLockObtainFailedException e) {
+            CompletableFuture<IndexShard> result = new CompletableFuture<>();
+            // ~1.1hours in total; Retries take never more than 5000ms
```

Comment:
minor: Isn't it ~1.4h?
`((1000 * 5000 + 50) / 3600000) ~= 1.388` 
```suggestion
            // ~1.4hours in total; Retries take never more than 5000ms
```

Reply:
The increments are different:

https://github.com/crate/crate/blob/99d183032d4b5763a24a84ddf909806d94df4081/server/src/main/java/io/crate/action/LimitedExponentialBackoff.java#L57-L71

I arrived at the 1.1 by summing up the values.